### PR TITLE
Make tidb comptible compatible with navicat

### PR DESCRIPTION
**Pull request by shenli**
Fix database list empty problem.
I will refactor buildShowSchema related code latter.

### CODE REVIEW

File: plan/planbuilder.go [hunk 1]
```diff
@@ -826,13 +826,39 @@ func buildShowEventsSchema() *expression.Schema {
...
+		retType.Charset, retType.Collate = types.DefaultCharsetForType(retType.Tp)
+		col.RetType = &retType
+		schema.Append(col)
+	}
+	return schema
+}
+
 // getShowColNamesAndTypes gets column names and types. If the `ftypes` is empty, every column is set to varchar type.
-func getShowColNamesAndTypes(s *ast.ShowStmt) (names []string, ftypes []byte) {
+// func getShowColNamesAndTypes(s *ast.ShowStmt) (names []string, tps []types.FieldType) {
```
Comment (hanfei1991) [inline plan/planbuilder.go]: update comments

Comment (coocood) [inline plan/planbuilder.go]: Update comment.

**Comment by shenli** [comment]
@coocood @hanfei1991 PTAL

File: plan/planbuilder.go [hunk 2]
```diff
@@ -826,13 +826,38 @@ func buildShowEventsSchema() *expression.Schema {
 	return schema
 }
 
-// getShowColNamesAndTypes gets column names and types. If the `ftypes` is empty, every column is set to varchar type.
```
Comment (coocood) [inline plan/planbuilder.go]: This function is only used by `buildShowSchema`, I think
`buildShowSchemaWithNamesAndTypes` is better.

Comment (shenli) [inline plan/planbuilder.go]: I will remove this function in the next PR.

**Comment by coocood** [comment]
LGTM

**Pull request merged**